lsp-fiber/src/rpc.mli

```diff
@@ -1,6 +1,12 @@
 open! Import
 open Jsonrpc
 
+module Cancel : sig
```

Comment:
If the `Cancel` module was hidden and we provided functions like `val register_cancel : 'a t -> (unit -> unit Fiber.t) -> unit`, we wouldn't need to have mutable and ref in the `Cancel`, would make it easier to reason about the flow.

Reply:
What's `'a t` here?

Reply:
`Rpc.S.t`, so that we could inline the API of `Cancel` in `Rpc.S`.

Reply:
Ah, but that doesn't work as that `'a t` isn't request specific. We'd need to introduce something like `Context.t` on the jsonrpc side. That's something that I've been itching to do, but didn't get around to.

I admit that using fiber local storage isn't very attractive, but it's the easiest way to proceed without large refactoring. I will get rid of the unnecessary `Cancel` public module and expose a single function.